### DISCUSSION TITLE
How to update the visibility status of a QDialog in a parent component?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#8849] [General - Components / Directives / etc] [answered]
I am writing a small note-taking program and at some point, I have a list of notes (in a component called `DisplayNotes`) , each of them has an "edit" button. When clicking this button, I open a`QDialog`  to edit some of the note data (in a component called `EditNote`).

## What works

The structure of the components is the following

![image](https://user-images.githubusercontent.com/1345886/114307988-85064880-9ae2-11eb-9642-3e14082850a5.png)

`MainLayout` is the coordinating top component that listens to events from `DisplayNotes`, toggles a "visibility variable", which drives the visibility of `EditNote`:

```
<template>
  <div class="q-pa-md">
    <display-notes />
    <edit-note :display="displayNoteEditor" :note_id="note_id" />
  </div>
</template>
<script>
import EditNote from "components/EditNote";
import DisplayNotes from "components/DisplayNotes";

export default {
  name: "MainLayout",
  components: {
    "edit-note": EditNote,
    "display-notes": DisplayNotes,
  },
  data() {
    return {
      displayNoteEditor: false,
      note_id: "",
    };
  },
  created() {},
  mounted() {
    this.$root.$on("display_note_editor", (display) => (this.displayNoteEditor = true));
    this.$root.$on("hide", (display) => (this.displayNoteEditor = false));
    this.$root.$on("note_id", (note_id) => (this.note_id = note_id));
  },
};
</script>
```

`EditNote` hold the `QDialog` (note the `v-model="display"` part):

```
<template>
  <q-dialog v-model="display" transition-hide="scale" transition-show="scale">
    <q-card style="width: 100vw; height: 100vh">
      <q-bar :class="savedBackground.color" class="text-white text-center col">
        {{ savedBackground.message }}
      </q-bar>
```

## What does not work

Once I close the `QDialog` I need somehow to toggle the value of `displayNoteEditor`  in `MainLayout` (so that a subsequent click on a note opens `NoteEdit`). **How can I do that?**

**Comment by metalsadman** [comment, marked as answer]
please a make a codepen. on first glance, i wouldn't use `$root.$on` here, and just use normal event listening in the component tag ie. `<display-notes @hide=".." />`, also take into account that `$on` is removed in vue 3. https://v3.vuejs.org/guide/migration/events-api.html#_2-x-syntax

**Reply by wsw70** [reply]
Thanks, in the meantime I did exactly this.
